addon/-private/system/schema.js

```diff
@@ -0,0 +1,295 @@
+import EmptyObject from './empty-object';
+import { assert, warn } from 'ember-data/-private/debug'
+import Ember from 'ember';
+
+const {
+  get
+} = Ember;
+
+export default class Schema {
+  constructor(modelClass, store) {
+    this.primaryKey = 'id';
+    this._attributes = null;
+    this._relationships = null;
+    this._relationshipMap = null;
+    this._properties = null;
+    this._expandedModel = false;
+    this.modelClass = modelClass;
+    this.modelName = modelClass.modelName;
+    this._recordMap = null;
+    this._inverseMap = null;
+    this._adapter = null;
+    this._serializer = null;
+    this._superclass = null;
+    this.store = store;
+  }
+
+  get superclass() {
+    if (this._superclass === null) {
+      let superclass = this.modelClass.superclass;
+
+      this._superclass = superclass ? this.store.schemaFor(superclass.modelName) : undefined;
+    }
+
+    return this._superclass;
+  }
+
+  get adapter() {
+    if (this._adapter === null) {
+      this._adapter = this.store.adapterFor(this.modelName);
+    }
+    return this._adapter;
+  }
+
+  get serializer() {
+    if (this._serializer === null) {
+      this._serializer = this.store.serializerFor(this.modelName);
+    }
+    if (!this._serializer) {
+      throw new Error('Cant find serializer!');
+    }
```

Comment:
I believe it's just a left over check for something that was failing for me early in the process and can be removed